### ISSUE TITLE
Tests fail

### ISSUE THREAD

**Issue by yurivict** [xtensor-stack/xsimd#917]
```
-- xsimd v11.0.0
-- Tests build type is Release
-- Configuring done
-- Generating done
-- Build files have been written to: /usr/ports/devel/xsimd/work/.build
ninja: no work to do.
[  0% 1/1] cd /usr/ports/devel/xsimd/work/.build && /usr/local/bin/ctest --force-new-ctest-process
Test project /usr/ports/devel/xsimd/work/.build
    Start 1: test_xsimd
1/1 Test #1: test_xsimd .......................***Failed    0.52 sec
[doctest] doctest version is "2.4.11"
[doctest] run with "--help" for options
===============================================================================
/usr/ports/devel/xsimd/work/xsimd-11.0.0/test/test_xsimd_api.cpp:695:
TEST CASE:  [xsimd api | float types functions]<float>
  exp10

/usr/ports/devel/xsimd/work/xsimd-11.0.0/test/test_xsimd_api.cpp:466: ERROR: CHECK_EQ( extract(xsimd::exp10(T(val))), std::pow(value_type(10), val) ) is NOT correct!
  values: CHECK_EQ( 100, 100 )

===============================================================================
/usr/ports/devel/xsimd/work/xsimd-11.0.0/test/test_xsimd_api.cpp:695:
TEST CASE:  [xsimd api | float types functions]<double>
  exp10

/usr/ports/devel/xsimd/work/xsimd-11.0.0/test/test_xsimd_api.cpp:466: ERROR: CHECK_EQ( extract(xsimd::exp10(T(val))), std::pow(value_type(10), val) ) is NOT correct!
  values: CHECK_EQ( 100, 100 )

===============================================================================
[doctest] test cases:  303 |  301 passed | 2 failed | 0 skipped
[doctest] assertions: 7929 | 7927 passed | 2 failed |
[doctest] Status: FAILURE!


0% tests passed, 1 tests failed out of 1

Total Test time (real) =   0.52 sec

The following tests FAILED:
	  1 - test_xsimd (Failed)
Errors while running CTest
FAILED: CMakeFiles/test.util 
```

Verison: 11.0.0
clang-14
FreeBSD 13.1

**Comment by serge-sans-paille** [comment]
Thanks for reporting. Can you tell us if #916 fixed the issue?

**Comment by serge-sans-paille** [comment]
@yurivict do you still encounter the issue with latest master branch?

**Comment by doronbehar** [comment]
Here on NixOS, we try to build 11.1.0 on MacOS, which includes #918, and it didn't fix the issue for us. See progress at https://github.com/NixOS/nixpkgs/pull/246200

**Comment by yurivict** [comment]
11.2.0 still has this problem.

**Comment by yurivict** [comment]
12.1.1 still has the problem:
```
$ ./work/.build/test/test_xsimd
[doctest] doctest version is "2.4.11"
[doctest] run with "--help" for options
===============================================================================
/usr/ports/devel/xsimd/work/xsimd-12.1.1/test/test_xsimd_api.cpp:747:
TEST CASE:  [xsimd api | float types functions]<float>
  exp10

/usr/ports/devel/xsimd/work/xsimd-12.1.1/test/test_xsimd_api.cpp:524: ERROR: CHECK_EQ( extract(xsimd::exp10(T(val))), std::pow(value_type(10), val) ) is NOT correct!
  values: CHECK_EQ( 100, 100 )

===============================================================================
/usr/ports/devel/xsimd/work/xsimd-12.1.1/test/test_xsimd_api.cpp:747:
TEST CASE:  [xsimd api | float types functions]<double>
  exp10

/usr/ports/devel/xsimd/work/xsimd-12.1.1/test/test_xsimd_api.cpp:524: ERROR: CHECK_EQ( extract(xsimd::exp10(T(val))), std::pow(value_type(10), val) ) is NOT correct!
  values: CHECK_EQ( 100, 100 )

===============================================================================
[doctest] test cases:  321 |  319 passed | 2 failed | 0 skipped
[doctest] assertions: 8239 | 8237 passed | 2 failed |
[doctest] Status: FAILURE!
```

**Comment by yurivict** [comment]
Tests are still broken in 13.0.0

**Comment by serge-sans-paille** [comment]
@yurivict which architecture?

**Comment by yurivict** [comment]
amd64

**Comment by serge-sans-paille** [comment]
and which os / compiler / libc ? Because we (obviously!) test on these arch...

**Comment by serge-sans-paille** [comment]
@yurivict does #917 do the the trick?

**Comment by serge-sans-paille** [comment]
and thanks @doronbehar , that did help

**Comment by serge-sans-paille** [comment]
I do :-)

**Comment by serge-sans-paille** [comment]
@doronbehar so, does it fix the bug?

**Comment by doronbehar** [comment]
Only now I had time to delve in that. :) If I understand correctly, #1021 is supposed to maybe help us only with two patches we currently apply:

- https://github.com/NixOS/nixpkgs/blob/8ee3dcb13c44ec3d72c338a504b08669a0e832f4/pkgs/development/libraries/xsimd/disable-exp10-test.patch
- https://github.com/NixOS/nixpkgs/blob/8ee3dcb13c44ec3d72c338a504b08669a0e832f4/pkgs/development/libraries/xsimd/fix-darwin-exp10-implementation.patch

I also noticed many projects in the `@xtensor-stack` had version updates, so I currently test the updates with our CI in https://github.com/NixOS/nixpkgs/pull/312745 . It will take time as this our CI is especially needed for Darwin tests, and for those platforms it is rather slow.

**Comment by doronbehar** [comment]
Ok so putting asside now #1024 (which (I hope) fixes #804), we didn't experience the failures reported here on any platform - including aarch64-darwin, for which it took our CI a few days to reach.

**Comment by serge-sans-paille** [comment]
So I think we can close this issue, feel free to reopen if CI proves me wrong.
